src/osdp_pd.c

```diff
@@ -1019,6 +1019,7 @@ static void osdp_pd_update(struct osdp_pd *pd)
 	ret = pd_receive_and_process_command(pd);
 
 	if (ret == OSDP_PD_ERR_IGNORE || ret == OSDP_PD_ERR_NO_DATA) {
+		osdp_phy_state_reset(pd, false);
```

Comment:
We cannot reset phy state for `OSDP_PD_ERR_NO_DATA`. Instead, this has to be moved to `pd_receive_and_process_command()` under `case OSDP_ERR_PKT_SKIP:`.